### ISSUE TITLE
JSON Output for Type::none Fields

### ISSUE THREAD

**Issue by amiremohamadi** [bpftrace/bpftrace#3543]
Currently, when a field of type `Type::none` is encountered, its value is not printed -- as we return an empty string
https://github.com/bpftrace/bpftrace/blob/c14a2ce8b7afa3950e7263d509bd869cb1284b97/src/output.cpp#L417-L419

This leads to invalid JSON formatting because the fields with missing values are not represented. 

It would be an easy fix to return `”null”` for json outputs in these cases.

**Comment by jordalgo** [comment]
@amiremohamadi Do you have an example script that outputs this?

**Comment by amiremohamadi** [comment]
for instance in this case:
```
$ bpftrace -e 'kprobe:vfs_write {print(*(struct file *)arg0); exit()}'
```
I see some missing values (E.g. `f_count`)
```
{ .f_llist = { .next = 0x0 }, .f_rcuhead = { .next = 0x0, .func = 0x0 }, .f_iocb_flags = 0, .f_lock = { .rlock = { .raw_lock = { .val = , .locked = 0, .pending = 0, .locked_pending = 0, .tail = 0 } } }, .f_mode = 135135239, .f_count = , .f_pos_lock = { .owner = , .wait_lock = { .raw_lock = { .val = , .locked = 0, .pending = 0, .locked_pending = 0, .tail = 0 } }, .osq = { .tail =  }, .wait_list = { .next = 0xffff931c675f6b30, .prev = 0xffff931c675f6b30 } }, .f_pos = 0, .f_flags = 2050, .f_owner = { .lock = , .pid = 0x0, .pid_type = 0, .uid = , .euid = , .signum = 0 }, .f_cred = 0xffff931b9f4a2180, .f_ra = { .start = 0, .size = 0, .async_size = 0, .ra_pages = 0, .mmap_miss = 0, .prev_pos = 0 }, .f_path = { .mnt = 0xffff931b8141c660, .dentry = 0xffff931bc3b1b900 }, .f_inode = 0xffff931b804fdb80, .f_op = 0xffffffff88287d80, .f_version = 0, .f_security = 0xffff931c0eb5ed50, .private_data = 0xffff931ba836e340, .f_ep = 0xffff931bc52eac10, .f_mapping = 0xffff931b804fdcf8, .f_wb_err = 0, .f_sb_err = 0 }
```
in this specific case, we should find a way to print the values since they aren't truly null. but i'm not sure about other scenarios, so IMO generally returning `"null"` in JSON outputs instead of empty strings would be reasonable.

@jordalgo

**Comment by jordalgo** [comment]
Yeah, I guess bpftrace's `None` type is equivalent to `NULL` in C. Not sure if we've discussed it much because we have a bunch of checks to prevent the use of the raw None type in a lot of cases. Let me think on it a bit more but printing "null" for JSON output sounds reasonable.

**Comment by danobi** [comment]
> I guess bpftrace's None type is equivalent to NULL in C.

I didn't look, but is that true? I'd have thought C NULL in bpftrace is a pointer type with value 0x0. 

I also thought there should not be any Type::none in AST past semantic analysis. To me Type::none was always tech debt for not using std::optional<SizedType>.

Should look into if we even expect none in types first.

**Comment by viktormalik** [comment]
Also we should make sure that whatever value we print here is a valid JSON value.

**Comment by jordalgo** [comment]
Fixed (ish) here: https://github.com/bpftrace/bpftrace/pull/3692
